tests/media_thumbnail_test.go

```diff
@@ -68,6 +73,88 @@ func TestRemotePngThumbnail(t *testing.T) {
 	}
 }
 
+func TestFederationThumbnail(t *testing.T) {
+	runtime.SkipIf(t, runtime.Dendrite)
+
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+
+	srv := federation.NewServer(t, deployment,
+		federation.HandleKeyRequests(),
+	)
+	cancel := srv.Listen()
+	defer cancel()
+	origin := spec.ServerName(srv.ServerName())
+
+	fileName := "test.png"
+	contentType := "image/png"
+
+	uri := alice.UploadContent(t, data.LargePng, fileName, contentType)
+	mediaOrigin, mediaId := client.SplitMxc(uri)
+
+	path := []string{"_matrix", "media", "v3", "thumbnail", mediaOrigin, mediaId}
+	res := alice.MustDo(t, "GET", path, client.WithQueries(url.Values{
+		"width":  []string{"32"},
+		"height": []string{"32"},
+		"method": []string{"scale"},
+	}))
+
+	if res.StatusCode != 200 {
+		t.Fatalf("thumbnail request for uploaded file failed")
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		t.Fatalf("thumbnail request for uploaded file failed: %s", err)
+	}
+
+	fedReq := fclient.NewFederationRequest(
+		"GET",
+		origin,
+		"hs1",
+		"/_matrix/federation/v1/media/thumbnail/"+mediaId+"?method=scale&width=32&height=32",
+	)
+
+	resp, err := srv.DoFederationRequest(context.Background(), t, deployment, fedReq)
+	if err != nil {
+		t.Fatalf("federation thumbnail request for uploaded file failed: %s", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("federation thumbnail request for uploaded file failed with status code: %v", resp.StatusCode)
+	}
+
+	resContentType := resp.Header.Get("Content-Type")
+	_, params, err := mime.ParseMediaType(resContentType)
+
+	if err != nil {
+		t.Fatalf("failed to parse multipart response: %s", err)
+	}
+
+	reader := multipart.NewReader(resp.Body, params["boundary"])
+	for {
+		p, err := reader.NextPart()
+		if err == io.EOF { // End of the multipart content
+			break
+		}
+		if err != nil {
+			t.Fatalf("failed to read multipart response: %s", err)
+		}
+
+		partContentType := p.Header.Get("Content-Type")
+		if partContentType == contentType {
+			imageBody, err := io.ReadAll(p)
+			if err != nil {
+				t.Fatalf("failed to read multipart part %s: %s", partContentType, err)
+			}
+			if !bytes.Equal(imageBody, body) {
+				t.Fatalf("body does not match uploaded file")
+			}
+		}
+	}
```

Comment:
Can we add a check after the loop to make sure that an image was actually included in the response, mostly to make it clear that we are *actually* checking the image. Just something like `have_checked_image = true; ...; assert have_changed_image`

Reply:
Right this is sorted - if you're happy with the change can you hit the merge button for me (as I no longer posses the power). Thanks so much!